### PR TITLE
Vps 109/add group member from UI

### PR DESCRIPTION
**Pull request by rgin216**
## Issue

Managing scenario group members required exporting the group CSV, manually adding a row, and importing it back into the application.

## Solution

Added a UI flow on the manage-groups page to add a single member directly from the application. This includes a modal for name, email, role, and group number, plus a backend endpoint that appends the member without rebuilding all groups. The endpoint validates required fields, duplicate emails, and duplicate roles within a group.

## Risk

Low to moderate. This adds a new group mutation path, so the main risk is inconsistent group/role data if validation misses an edge case. Existing CSV import behaviour is unchanged.

## Checklist

- [ ] Acceptance criteria met
- [ ] Wiki documentation is written and up to date
- [x] Unit tests written and passing
- [ ] Integration tests written and passing
- [ ] Continuous integration build passing

## Summary by CodeRabbit

* **New Features**
  * Added "Add Member" modal and button to add group members (name, email, role, group number) with trimming/validation and submit state.

* **Bug Fixes**
  * Backend now robustly handles concurrent member additions and legacy group records to prevent duplicates and ensure members are added reliably.
  * Role/email duplicate checks are case/whitespace-insensitive.

* **Tests**
  * Expanded API tests, including a concurrency case verifying single-group creation with multiple simultaneous adds.

### CODE REVIEW

File: backend/src/db/models/group.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ const groupSchema = new Schema({
   scenarioId: {
     type: String,
   },
+  group: {
+    type: String,
+  },
```
Comment (harbassan) [inline backend/src/db/models/group.js:21]: i dont quite understand this field and the logic associated with it . what does the "group" of the group actually mean?

**Comment by rgin216** [comment]
Changed to be in dashboard not in a scenario
